### ISSUE TITLE
It seems dotnet_diagnostic.IDE0058.severity in .editorconfig started to be ignored with today's VS update (13 feb 2024)

### ISSUE THREAD

**Issue by dggmez** [dotnet/roslyn#72094]
**Version Used**: 

VS 2022 version 17.9.0. (MSBuild version seems to be 17.9.4+90725d08d for .NET)

**Steps to Reproduce**:

1. Add `dotnet_diagnostic.IDE0058.severity = none` to .editorconfig
2. Build solution
3. See how "Expression value is never used" issues are reported

Today I updated VS 2022 to version 17.9.0. It seems that the configuration for rule IDE0058 is being ignored. Not only the issues are reported even though I set the severity as `none` but even if I set it as `error` the code compiles just fine. We have SonarQube running for our builds and now the PR is full of comments saying "Bug (info): Expression value is never used (external_roslyn:IDE0058)". 

**Diagnostic Id**:

IDE0058

**Expected Behavior**:

Ignore rule just like it was being done in the previous versions.

**Actual Behavior**:

Issues with code IDE0058 are being reported with severity = info.

**Comment by dggmez** [comment]
OK. I think I finally figured it out. Previously you could have this:

    csharp_style_unused_value_expression_statement_preference = discard_variable:suggestion
    dotnet_diagnostic.IDE0058.severity = none

But from today's version it will report IDE0058 issues if you don't set the `csharp_style_unused_value_expression_statement_preference ` option as `nonè. So I fixed it by changing the previous two lines to this:

    csharp_style_unused_value_expression_statement_preference = discard_variable:none
    dotnet_diagnostic.IDE0058.severity = none

So I don't know if this is a bug in the new version or a fix of a previous bug. 🤔

**Comment by sid-6581** [comment]
> @sid-6581 Yeah but it seems that's how it used to work in previous versions or _at least_ I can say with certainty that was the case for rule IDE0058.

Yes, I had a few mismatches, but in 8.0.101 and earlier, the diagnostic severity always won.

**Comment by adamhathcock** [comment]
This type of thing has bit me for a number of rules in SDK 8.0.201

I fixed most issues by changing my default from suggestion:

```
dotnet_analyzer_diagnostic.severity = silent
```

**Comment by mavasani** [comment]
Please see https://github.com/dotnet/roslyn/issues/52991#issuecomment-1960692440 for details on this regression and suggested approach to workaround it.

**Comment by Tiberriver256** [comment]
Just adding a note that this doesn't seem specifically tied to Visual Studio. I have the same problem when using `dotnet build` when the SDK version is `8.0.200` or newer.

Opened an issue in the SDK repo here: https://github.com/dotnet/sdk/issues/38980

I did start seeing the error when I updated Visual Studio, but that was because Visual Studio updated my .NET Core SDK version, and I can easily avoid the problem by updating my `global.json` to use the `8.0.102` version of the SDK.

**Comment by sharwell** [comment]
> OK. I think I finally figured it out. Previously you could have this in your .editorconfig:
> 
> ```
> csharp_style_unused_value_expression_statement_preference = discard_variable:suggestion
> dotnet_diagnostic.IDE0058.severity = none
> ```
> 
> But from today's version it will report IDE0058 issues if you don't set the `csharp_style_unused_value_expression_statement_preference ` option as `none`. So I fixed it by changing the previous two lines to this:
> 
> ```
> csharp_style_unused_value_expression_statement_preference = discard_variable:none
> dotnet_diagnostic.IDE0058.severity = none
> ```
> 
> So I don't know if this is a bug in the new version or a fix of a previous bug. 🤔

Hi @dggmez, I'm reviewing this in more detail and have a couple questions for you:

1. In following the second of your repro steps "Build solution", I would not expect the value `discard_variable:suggestion` to result in any diagnostic showing up as part of a build. Are you sure the value wasn't set to `discard_variable:warning` or `discard_variable:error`?
2. The severity setting for `csharp_style_unused_value_expression_statement_preference` is optional. Can you confirm that changing it to just `discard_variable` allows the system to use the severity specified in `dotnet_diagnostic.IDE0058.severity`?

Thanks!
